src/codegen/math/float/tanh.rs

```diff
@@ -0,0 +1,117 @@
+//! Vertical floating-point `tanh`
+#![allow(unused)]
+
+// FIXME 64-bit 1 elem vectors tanh
+
+use crate::*;
+
+crate trait Tanh {
+    fn tanh(self) -> Self;
+}
+
+macro_rules! define_tanh {
+
+    ($name:ident, $basetype:ty, $simdtype:ty, $lanes:expr, $trait:path) => {
+        fn $name(x: $simdtype) -> $simdtype {
+            let mut buf = [0.0; $lanes];
+            x.write_to_slice_unaligned(&mut buf);
```

Comment:
Instead of using `write_to_slice` here, I think it would be better to just do a `transmute`:

```rust
let mut buf: [$base_type; $lanes] = unsafe { mem::transmute(buf) };
// ...
unsafe { transmute(buf) }
```

since arrays and simd vectors are layout-compatible.